### PR TITLE
Hb 07

### PR DESCRIPTION
**Pull request by vlykhvar**
1. Create models
- Order

2. Create DAO
- OrderDao

3. Create service
- OrderService

### CODE REVIEW

**Review by Likh-Alex — APPROVED** [review]

File: src/main/java/com/dev/theater/Main.java [hunk 1]
```diff
@@ -55,11 +59,15 @@ public static void main(String[] args) throws AuthenticationException {
         authenticationService.register("test2@bigmir.net", "12345");
         System.out.println(authenticationService.login("test2@bigmir.net", "12345").toString());
         userService.getAll().forEach(System.out::println);
-        ShoppingCart shoppingCart = shoppingCartService.getByUser(user);
         shoppingCartService.addSession(movieSession, user);
+        Ticket ticket = new Ticket();
+        ticket.setUser(user);
+        ticket.setMovieSession(movieSession);
+        ShoppingCart shoppingCart = shoppingCartService.getByUser(user);
+        shoppingCart.getTicketList().add(ticket);
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/theater/Main.java]: let's remove this and use `addSession()` instead

File: src/main/java/com/dev/theater/dao/impl/OrderDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+import java.util.List;
+import org.hibernate.Session;
+
+@Dao
+public class OrderDaoImpl extends DaoImpl<Order> implements OrderDao {
+    @Override
+    public List<Order> getOrdersHistory(User user) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            return session.createQuery("from Order o inner join fetch o.tickets"
+                    + " where o.user = :user", Order.class)
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/theater/dao/impl/OrderDaoImpl.java]: please see this conversation
https://mateacademy.slack.com/archives/G01EHB8CWAE/p1612455330236200

File: src/main/java/com/dev/theater/model/Order.java [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+    public void setId(Long id) {
+        this.id = id;
+    }
+
+    public LocalDateTime getOrderTime() {
+        return orderTime;
+    }
+
+    public void setOrdetTime(LocalDateTime orderTime) {
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/theater/model/Order.java]:
```suggestion
    public void setOrderTime(LocalDateTime orderTime) {
```

File: src/main/java/com/dev/theater/model/Order.java [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    public void setUser(User user) {
+        this.user = user;
+    }
+
+    @Override
+    public String toString() {
+        return "Order{"
+                + "id=" + id
+                + ", ticets=" + tickets
+                + ", ordetTime=" + orderTime
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/theater/model/Order.java]:
```suggestion
                + ", tickets=" + tickets
                + ", orderTime=" + orderTime
```

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/com/dev/theater/dao/impl/OrderDaoImpl.java (hunk 2 above)
Comment (vlykhvar) [reply src/main/java/com/dev/theater/dao/impl/OrderDaoImpl.java]: changed

File: src/main/java/com/dev/theater/model/Order.java (hunk 4 above)
Comment (vlykhvar) [reply src/main/java/com/dev/theater/model/Order.java]: done

File: src/main/java/com/dev/theater/model/Order.java (hunk 3 above)
Comment (vlykhvar) [reply src/main/java/com/dev/theater/model/Order.java]: done

File: src/main/java/com/dev/theater/Main.java (hunk 1 above)
Comment (vlykhvar) [reply src/main/java/com/dev/theater/Main.java]: done

File: src/main/java/com/dev/theater/Main.java [hunk 5]
```diff
@@ -55,11 +59,15 @@ public static void main(String[] args) throws AuthenticationException {
         authenticationService.register("test2@bigmir.net", "12345");
         System.out.println(authenticationService.login("test2@bigmir.net", "12345").toString());
         userService.getAll().forEach(System.out::println);
+        shoppingCartService.addSession(movieSession, user);
+        Ticket ticket = new Ticket();
+        ticket.setUser(user);
+        ticket.setMovieSession(movieSession);
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/theater/Main.java]: we can remove this, right?

**Review by KseniiaMakarova — APPROVED** [review]

File: src/main/java/com/dev/theater/Main.java (hunk 5 above)
Comment (vlykhvar) [reply src/main/java/com/dev/theater/Main.java]: done

**Pull request merged**